common/oatbox/service/ServiceManager.php

```diff
@@ -108,6 +108,11 @@ private function load($serviceId, $serviceKey)
         if ($service === false) {
             $service = $this->tryAutowiring($serviceId, $serviceKey);
         }
+
+        if ($service instanceof ServiceFactoryInterface) {
+            $service = $service($this);
```

Comment:
Aha, so you register a factory, but when you call the service locator with its ID, you're gonna receive an actual service, made by that factory. Tricky, and may potentially be very misleading.